### PR TITLE
Update the bindgen derive logic

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Previously many generated types were missing common derive traits like
"Clone", "Debug", etc.  Now most types default to having those traits.
There is a separate list for:
- Types that are enums
- Types that should implement the "Copy" trait.
- Types that are dynamically sized (These implement no traits)

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #91** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/91" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/91?utm_source=stack-comment).

**Review by awygle — APPROVED** [review]

File: core/build/src/lib.rs
Comment (jcape) [inline core/build/src/lib.rs:98]:
```suggestion
    pub fn enum_types<'a, E, I>(enum_types: I) -> Self
```

File: core/sys/types/build.rs
Comment (jcape) [inline core/sys/types/build.rs:55]:
```suggestion
    let callback = SgxParseCallbacks::default()
        .enum_types(["sgx_status_t", "sgx_quote_sign_type_t"].iter())
```

File: dcap/sys/types/build.rs
Comment (jcape) [inline dcap/sys/types/build.rs:41]: Can't change this in a GH comment, but having this be `default().enum_types([ ... ])` would be the goal

File: tcrypto/sys/types/build.rs
Comment (jcape) [inline tcrypto/sys/types/build.rs:38]: Same: `default().enum_types()`

File: tservice/sys/types/build.rs
Comment (jcape) [inline tservice/sys/types/build.rs:32]: Same: `::default().enum_types()`

**Review by jcape — APPROVED** [review]
LGTM, but IMO `enum_types` should follow the same pattern as the other "special" arrays.

File: core/build/src/lib.rs
Comment (nick-mobilecoin) [reply core/build/src/lib.rs]: done

File: core/sys/types/build.rs
Comment (nick-mobilecoin) [reply core/sys/types/build.rs]: done

File: tcrypto/sys/types/build.rs
Comment (nick-mobilecoin) [reply tcrypto/sys/types/build.rs]: done

File: tservice/sys/types/build.rs
Comment (nick-mobilecoin) [reply tservice/sys/types/build.rs]: done

File: dcap/sys/types/build.rs
Comment (nick-mobilecoin) [reply dcap/sys/types/build.rs]: done

**Comment by nick-mobilecoin** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/91?utm_source=stack-comment).


Job Reference: 2v9KzWSc3RZmb6aSw9co

**Pull request merged**
